wagtail/contrib/modeladmin/tests/test_simple_modeladmin.py

```diff
@@ -758,9 +758,12 @@ def test_inspect_view(self):
         # and only does one query per modeladmin
         with CaptureQueriesContext(connection) as queries:
```

Comment:
I would be happy with a simple `with self.assertNumQueries(3):` here (or whatever the number of queries is after this optimisation) with a comment to explain why that is the expectation. Really, we just want to keep an eye on that number - if it jumps up for any reason, it's a good sign that something is wrong.

Reply:
How is the number of expected queries determined? 

From the CI builds it seems to vary based on the configuration. I'm a bit confused as to why it's not seeing any permissions queries in a few of the CI runs.

Reply:
I would start with `self.assertNumQueries(3)`, then see how you get on -> it fail nice and clearly when the number of queries does not match. If there is still inconsistency between builds, it might be something silly, like content types being cached some times and not others.

Reply:
@frmdstryr Okay, so it looks like it's consistently 35 or 25. So, I guess we just need to look at those queries more closely and figure out what those extra 10 queries are, and where they are coming from. It could just be a case of making sure a cache is populated before loading the list, for example: `Site. get_site_root_paths()` (which is used in URL generation)